### DISCUSSION TITLE
Best practice to fail a CD pipeline when a rollout isn't successful

### DISCUSSION THREAD

**Discussion by novatan-rb** [argoproj/argo-rollouts#996] [Q&A] [answered]
Hi everyone,

As the title suggested, I'm wondering what's the best practice to fail a CD pipeline when a rollout isn't successful.

Use case: a build goes red when the rollout is rolled back and it's in degraded state.

For kubernetes native deployments you can do it easily with `kubectl rollout status`, but unfortunately that doesn't work for CRDs (https://github.com/kubernetes/kubectl/issues/690). I know I could whip up a little bash script to check the status in a while loop. Just wondering if a more elegant approach exists?

Thanks!

**Comment by kostis-codefresh** [comment, marked as answer]
Would the rollout status command work in your case? https://argoproj.github.io/argo-rollouts/generated/kubectl-argo-rollouts/kubectl-argo-rollouts_status/

**Reply by novatan-rb** [reply]
Hi @kostis-codefresh, thanks for the response. This is exactly what I need. From the look of it, the `status` subcommand is not available yet in current latest version (0.10.2) of kubectl-argo-rollouts?

**Reply by kostis-codefresh** [reply]
Yes you are correct. Last official release was in December 2020. I am not sure when the next version will come (I am not one of the maintainers).

**Reply by kostis-codefresh** [reply]
@novatan-rb It seems that you can get more recent binaries from here https://github.com/orgs/argoproj/packages?repo_name=argo-rollouts

**Reply by novatan-rb** [reply]
Ah cool, thanks for the help.

**Reply by jessesuen** [reply]
v1.0 is [released](https://github.com/argoproj/argo-rollouts/releases/tag/v1.0.0)!
